eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -2,7 +2,7 @@
 . "${PSScriptRoot}\logging.ps1"
 . "${PSScriptRoot}\SemVer.ps1"
 
-$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+.*(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?)(\s+(?<releaseStatus>\(.*\)))?)"
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\##\s+(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?)(\s+(?<releaseStatus>\(.*\)))?)"
```

Comment:
```suggestion
$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.*\))))"
```
Lets use the semver regex and lets not make the release status optional, as I don't think there is any reason it should not be there for the entries we care to find. Of course please do some testing of this change by running it over the changelogs in our repos. Maybe even getting a before and after diff of what we find in our the logs to ensure it excludes the things we expect it to.

Reply:
If we make don't make release status optional won't we break older entries with no release status? We always need to parse all the entries so we care about all of them. I will test it with the SemVer REGEX.

Reply:
Why do we care about parsing older entries? If they appear as just content from the oldest entry that should work for the most part? I believe we only care about parsing entries for versions we are taking action on.

Reply:
Right now we parse all the entries, add new entries, sort the entries then write it back.

Reply:
So what happens if there is text at the top that doesn't have a header? I assume it is just left along. If there are headers that don't match at the bottom then they essentially will become the content of the lowest header that does match so they will essentially be carried with that version header and as long as those are older entries that we aren't working on I think that is probably ok.

Reply:
So far I haven't run into any of those cases. Generally we have been consistent to only have content under level 2 headers. Only the level 1 header  which is the title of the file lies outside. Consistent with our guideline everything under a level 2 header is considered a changelog entry till the next level 2 header. So far there has been no problem with this because everyone has just followed that pattern.